services/wireguard/endpoint/kernelspace/client.go

```diff
@@ -120,21 +126,17 @@ func addDefaultRoute(iface string) error {
 }
 
 func (c *client) Close() error {
-	d, err := c.wgClient.Device(c.iface)
-	if err != nil || d.Name != c.iface {
-		return err
-	}
+	destroyErr := c.DestroyDevice(c.iface)
+	closeErr := c.wgClient.Close()
 
-	if len(d.Peers) > 0 && d.Peers[0].Endpoint != nil {
-		if err := utils.SudoExec("ip", "route", "del", d.Peers[0].Endpoint.IP.String()); err != nil {
-			return err
-		}
-	}
-
-	if err := utils.SudoExec("ip", "link", "del", "dev", c.iface); err != nil {
-		return err
+	if closeErr != nil && destroyErr == nil {
+		return closeErr
+	} else if closeErr == nil && destroyErr != nil {
+		return destroyErr
+	} else if closeErr != nil && destroyErr != nil {
+		return fmt.Errorf("failed to close wireguard client %v; failed to destroy wireguard device: %v", closeErr, destroyErr)
```

Comment:
All these checks looks abundant for me. What if we had another function returning another error here? Would you cross-check all options too? 
I think we could fail fast (return) on first error and defer rest closes as they are.

Reply:
It was working this way initially, changed it after this comment: https://github.com/mysteriumnetwork/node/pull/608#discussion_r241315944

Reply:
oh, got your point about defering rest closes. Changed it.